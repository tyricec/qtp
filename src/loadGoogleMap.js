import util from './util'

class GMapScriptAlreadyLoadedError extends Error {
  constructor(message) {
    message = 'GMapScriptAlreadyLoaded: '.concat(message)
    super(message)
  }
}

function loadGoogleMap () {
  return new Promise((resolve, reject) => {
    var gMapSrc = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyA1_2TtftN8qv6qI-NYBQ9N7izoh9ps1YM&callback=qtp.initGMap'
    if (document.querySelector(`script[src="${gMapSrc}"]`)) {
      reject(new GMapScriptAlreadyLoadedError('Google Maps script tag was already added to the dom'))
    }
    let gScript = util.buildHTML('script', {
      src: gMapSrc,
    })

    window.qtp = {
      initGMap: onLoad,
    }

    document.body.appendChild(gScript)

    function onLoad() {
      resolve(window.google.maps)
    }
  })
}

export default loadGoogleMap