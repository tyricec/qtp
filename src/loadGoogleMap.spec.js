import loadGoogleMap from './loadGoogleMap'

beforeEach(() => {
  var gMapSrc = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyA1_2TtftN8qv6qI-NYBQ9N7izoh9ps1YM&callback=qtp.initGMap'
  var script = document.querySelector(`script[src="${gMapSrc}"]`)
  if (script) {
    document.body.removeChild(script)
  }
})

afterEach(() => {
  var gMapSrc = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyA1_2TtftN8qv6qI-NYBQ9N7izoh9ps1YM&callback=qtp.initGMap'
  var script = document.querySelector(`script[src="${gMapSrc}"]`)
  if (script) {
    document.body.removeChild(script)
  }
})

test('loadGoogleMap loads google maps', () => {
  return loadGoogleMap().then((gmap) => {
    expect(gmap).toMatchSnapshot()
  })
})

test('loadGoogleMap throws error when attempting to load twice', () => {
  loadGoogleMap()
  return loadGoogleMap().catch((e) => { 
    expect(/GMapScriptAlreadyLoaded/.test(e))
  })
})