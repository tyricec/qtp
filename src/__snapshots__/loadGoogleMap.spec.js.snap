// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadGoogleMap loads google maps 1`] = `
Object {
  "Animation": Object {
    "BOUNCE": 1,
    "DROP": 2,
    "lo": 4,
    "no": 3,
  },
  "BicyclingLayer": [Function],
  "Circle": [Function],
  "ControlPosition": Object {
    "BOTTOM": 11,
    "BOTTOM_CENTER": 11,
    "BOTTOM_LEFT": 10,
    "BOTTOM_RIGHT": 12,
    "CENTER": 13,
    "LEFT": 5,
    "LEFT_BOTTOM": 6,
    "LEFT_CENTER": 4,
    "LEFT_TOP": 5,
    "RIGHT": 7,
    "RIGHT_BOTTOM": 9,
    "RIGHT_CENTER": 8,
    "RIGHT_TOP": 7,
    "TOP": 2,
    "TOP_CENTER": 2,
    "TOP_LEFT": 1,
    "TOP_RIGHT": 3,
  },
  "Data": [Function],
  "DirectionsRenderer": [Function],
  "DirectionsService": [Function],
  "DirectionsStatus": Object {
    "INVALID_REQUEST": "INVALID_REQUEST",
    "MAX_WAYPOINTS_EXCEEDED": "MAX_WAYPOINTS_EXCEEDED",
    "NOT_FOUND": "NOT_FOUND",
    "OK": "OK",
    "OVER_QUERY_LIMIT": "OVER_QUERY_LIMIT",
    "REQUEST_DENIED": "REQUEST_DENIED",
    "UNKNOWN_ERROR": "UNKNOWN_ERROR",
    "ZERO_RESULTS": "ZERO_RESULTS",
  },
  "DirectionsTravelMode": Object {
    "BICYCLING": "BICYCLING",
    "DRIVING": "DRIVING",
    "TRANSIT": "TRANSIT",
    "WALKING": "WALKING",
  },
  "DirectionsUnitSystem": Object {
    "IMPERIAL": 1,
    "METRIC": 0,
  },
  "DistanceMatrixElementStatus": Object {
    "NOT_FOUND": "NOT_FOUND",
    "OK": "OK",
    "ZERO_RESULTS": "ZERO_RESULTS",
  },
  "DistanceMatrixService": [Function],
  "DistanceMatrixStatus": Object {
    "INVALID_REQUEST": "INVALID_REQUEST",
    "MAX_DIMENSIONS_EXCEEDED": "MAX_DIMENSIONS_EXCEEDED",
    "MAX_ELEMENTS_EXCEEDED": "MAX_ELEMENTS_EXCEEDED",
    "OK": "OK",
    "OVER_QUERY_LIMIT": "OVER_QUERY_LIMIT",
    "REQUEST_DENIED": "REQUEST_DENIED",
    "UNKNOWN_ERROR": "UNKNOWN_ERROR",
  },
  "ElevationService": [Function],
  "ElevationStatus": Object {
    "INVALID_REQUEST": "INVALID_REQUEST",
    "OK": "OK",
    "OVER_QUERY_LIMIT": "OVER_QUERY_LIMIT",
    "REQUEST_DENIED": "REQUEST_DENIED",
    "UNKNOWN_ERROR": "UNKNOWN_ERROR",
    "ho": "DATA_NOT_AVAILABLE",
  },
  "FusionTablesLayer": [Function],
  "Geocoder": [Function],
  "GeocoderLocationType": Object {
    "APPROXIMATE": "APPROXIMATE",
    "GEOMETRIC_CENTER": "GEOMETRIC_CENTER",
    "RANGE_INTERPOLATED": "RANGE_INTERPOLATED",
    "ROOFTOP": "ROOFTOP",
  },
  "GeocoderStatus": Object {
    "ERROR": "ERROR",
    "INVALID_REQUEST": "INVALID_REQUEST",
    "OK": "OK",
    "OVER_QUERY_LIMIT": "OVER_QUERY_LIMIT",
    "REQUEST_DENIED": "REQUEST_DENIED",
    "UNKNOWN_ERROR": "UNKNOWN_ERROR",
    "ZERO_RESULTS": "ZERO_RESULTS",
  },
  "GroundOverlay": [Function],
  "ImageMapType": [Function],
  "InfoWindow": [Function],
  "KmlLayer": [Function],
  "KmlLayerStatus": Object {
    "DOCUMENT_NOT_FOUND": "DOCUMENT_NOT_FOUND",
    "DOCUMENT_TOO_LARGE": "DOCUMENT_TOO_LARGE",
    "FETCH_ERROR": "FETCH_ERROR",
    "INVALID_DOCUMENT": "INVALID_DOCUMENT",
    "INVALID_REQUEST": "INVALID_REQUEST",
    "LIMITS_EXCEEDED": "LIMITS_EXECEEDED",
    "OK": "OK",
    "TIMED_OUT": "TIMED_OUT",
    "UNKNOWN": "UNKNOWN",
  },
  "LatLng": [Function],
  "LatLngBounds": [Function],
  "MVCArray": [Function],
  "MVCObject": [Function],
  "Map": [Function],
  "MapTypeControlStyle": Object {
    "DEFAULT": 0,
    "DROPDOWN_MENU": 2,
    "HORIZONTAL_BAR": 1,
    "INSET": 3,
    "INSET_LARGE": 4,
  },
  "MapTypeId": Object {
    "HYBRID": "hybrid",
    "ROADMAP": "roadmap",
    "SATELLITE": "satellite",
    "TERRAIN": "terrain",
  },
  "MapTypeRegistry": [Function],
  "Marker": [Function],
  "MarkerImage": [Function],
  "MaxZoomService": [Function],
  "MaxZoomStatus": Object {
    "ERROR": "ERROR",
    "OK": "OK",
  },
  "NavigationControlStyle": Object {
    "ANDROID": 2,
    "DEFAULT": 0,
    "Dj": 5,
    "SMALL": 1,
    "ZOOM_PAN": 3,
    "oo": 4,
  },
  "OverlayView": [Function],
  "Point": [Function],
  "Polygon": [Function],
  "Polyline": [Function],
  "Rectangle": [Function],
  "SaveWidget": [Function],
  "ScaleControlStyle": Object {
    "DEFAULT": 0,
  },
  "Size": [Function],
  "StreetViewCoverageLayer": [Function],
  "StreetViewPanorama": [Function],
  "StreetViewPreference": Object {
    "BEST": "best",
    "NEAREST": "nearest",
  },
  "StreetViewService": [Function],
  "StreetViewSource": Object {
    "DEFAULT": "default",
    "OUTDOOR": "outdoor",
  },
  "StreetViewStatus": Object {
    "OK": "OK",
    "UNKNOWN_ERROR": "UNKNOWN_ERROR",
    "ZERO_RESULTS": "ZERO_RESULTS",
  },
  "StrokePosition": Object {
    "CENTER": 0,
    "INSIDE": 1,
    "OUTSIDE": 2,
  },
  "StyledMapType": [Function],
  "SymbolPath": Object {
    "BACKWARD_CLOSED_ARROW": 3,
    "BACKWARD_OPEN_ARROW": 4,
    "CIRCLE": 0,
    "FORWARD_CLOSED_ARROW": 1,
    "FORWARD_OPEN_ARROW": 2,
  },
  "TrafficLayer": [Function],
  "TrafficModel": Object {
    "BEST_GUESS": "bestguess",
    "OPTIMISTIC": "optimistic",
    "PESSIMISTIC": "pessimistic",
  },
  "TransitLayer": [Function],
  "TransitMode": Object {
    "BUS": "BUS",
    "RAIL": "RAIL",
    "SUBWAY": "SUBWAY",
    "TRAIN": "TRAIN",
    "TRAM": "TRAM",
  },
  "TransitRoutePreference": Object {
    "FEWER_TRANSFERS": "FEWER_TRANSFERS",
    "LESS_WALKING": "LESS_WALKING",
  },
  "TravelMode": Object {
    "BICYCLING": "BICYCLING",
    "DRIVING": "DRIVING",
    "TRANSIT": "TRANSIT",
    "WALKING": "WALKING",
  },
  "UnitSystem": Object {
    "IMPERIAL": 1,
    "METRIC": 0,
  },
  "ZoomControlStyle": Object {
    "DEFAULT": 0,
    "Dj": 3,
    "LARGE": 2,
    "SMALL": 1,
  },
  "__gjsload__": [Function],
  "event": Object {
    "Ka": [Function],
    "Rm": [Function],
    "T": [Function],
    "Zh": [Function],
    "addDomListener": [Function],
    "addDomListenerOnce": [Function],
    "addListener": [Function],
    "addListenerOnce": [Function],
    "bind": [Function],
    "clearInstanceListeners": [Function],
    "clearListeners": [Function],
    "forward": [Function],
    "hasListeners": [Function],
    "removeListener": [Function],
    "trigger": [Function],
  },
  "places": Object {
    "Autocomplete": [Function],
    "AutocompleteService": [Function],
    "PlacesService": [Function],
    "PlacesServiceStatus": Object {
      "INVALID_REQUEST": "INVALID_REQUEST",
      "NOT_FOUND": "NOT_FOUND",
      "OK": "OK",
      "OVER_QUERY_LIMIT": "OVER_QUERY_LIMIT",
      "REQUEST_DENIED": "REQUEST_DENIED",
      "UNKNOWN_ERROR": "UNKNOWN_ERROR",
      "ZERO_RESULTS": "ZERO_RESULTS",
    },
    "RankBy": Object {
      "DISTANCE": 1,
      "PROMINENCE": 0,
    },
    "RatingLevel": Object {
      "EXCELLENT": 2,
      "EXTRAORDINARY": 3,
      "GOOD": 0,
      "VERY_GOOD": 1,
    },
    "SearchBox": [Function],
  },
  "version": "3.28.14",
}
`;
